backend/utils/error_handler.py

```
@@ -0,0 +1,207 @@
+        logger.error(f"500 error: {str(error)}", exc_info=True)
+        from models import db
+        db.session.rollback()
+        return jsonify({'error': 'Internal server error'}), 500
+
+    @app.errorhandler(Exception)
+    def handle_exception(e):
+        logger.error(f"Unhandled exception: {str(e)}", exc_info=True)
+        return create_error_response(e, 500)
+
+
+def log_security_event(event_type, details, user_id=None, ip_address=None):
```

Ruff (0.11.9)

Redefinition of unused `log_security_event` from line 13

(F811)